src/backend/distributed/commands/schema.c

```diff
@@ -80,14 +80,14 @@ PostprocessCreateSchemaStmt(Node *node, const char *queryString)
 
 	if (ShouldUseSchemaBasedSharding(createSchemaStmt->schemaname))
 	{
-		/* for now, we don't allow creating tenant tables when creating the schema itself */
+		/*
+		 * We will call `citus_schema_distribute` after postprocess if it creates any
```

Comment:
When the syntax does not create any table, calling `citus_distribute_schema` throws error since the schema is propagated by different transaction so we should not call `citus_distribute_schema` if the syntax does not create any table. Instead we should propagate the schema at postprocess.

We can get rid of  propagation logic at postprocess if we let `citus_distribute_schema` propagate the schema at the start of its execution.